spec/facebook_ads/facebook_ads_spec.rb

```diff
@@ -1,8 +1,53 @@
+# frozen_string_literal: true
 require 'spec_helper'
 
 describe FacebookAds do
-  it 'indicates a version' do
-    # N.B. only a fake spec for the initial stub
-    expect(described_class::VERSION).to eql '0.1.0'
+  before do
+    described_class.send(:reset_configuration!)
+  end
+
+  describe '.configure' do
+    it 'passes the configuration as block parameter' do
+      described_class.configure do |config|
+        expect(config).to be described_class.configuration
+      end
+    end
+
+    it 'resets the client after configuration' do
+      described_class.configure do |config|
+        config.access_token = '1'
+      end
+
+      old_client = described_class.client
+
+      described_class.configure do |config|
+        config.access_token = '2'
+      end
+
+      expect(described_class.client).not_to be old_client
+    end
+  end
+
+  describe '.client' do
+    subject { described_class.client }
+
+    it 'creates a client using the configuration' do
+      expect(FacebookAds::Client)
+        .to receive(:new).with(described_class.configuration)
+      subject
+    end
+
+    it 'creates a client using the access token' do
+      client = double(FacebookAds::Client)
+      expect(FacebookAds::Client).to receive(:new).and_return(client)
+      is_expected.to be client
+    end
+
+    it 'creates the client only once' do
+      expect(FacebookAds::Client).to receive(:new).once.and_return(double)
+
+      described_class.client
```

Comment:
I should have probably added a comment... (will do that now)

I can't use `subject`, since RSpec caches the result of `subject`. That means the spec would always succeed, since we would never make the call twice (because `RSpec` only calls the block once).